File: RadioButton/app/src/main/java/com/example/radiobutton/Oyente.java
package com.example.radiobutton;

import android.text.Editable;
import android.text.TextWatcher;
import android.widget.RadioGroup;

public class Oyente implements RadioGroup.OnCheckedChangeListener, TextWatcher {

    private MainActivity activity;

    public Oyente(MainActivity activity) {
        this.activity = activity;
    }

    @Override
    public void onCheckedChanged(RadioGroup group, int checkedId) {
        activity.calcularValores();
    }

    @Override
    public void onTextChanged(CharSequence s, int start, int before, int count) {
        activity.calcularValores();
    }

    // DOS METODOS OBLIGATORIOS DE INTERFACE TEXTWATCHER, COMO NO LOS USARE NO SE DESAROLLAN
    @Override
    public void beforeTextChanged(CharSequence s, int start, int count, int after) {
    }

    @Override
    public void afterTextChanged(Editable s) {
    }
}

File: 04_RadioButton_kt/app/src/main/java/com/example/radiobuttonkt/Oyente.kt
package com.example.radiobuttonkt

import android.text.Editable
import android.text.TextWatcher
import android.widget.RadioGroup


class Oyente(private val activity: MainActivity) :
    RadioGroup.OnCheckedChangeListener,
    TextWatcher {

    override fun onCheckedChanged(group: RadioGroup, checkedId: Int) {
        activity.calcularValores()
    }

    override fun onTextChanged(s: CharSequence, start: Int, before: Int, count: Int) {
        activity.calcularValores()
    }

    override fun beforeTextChanged(s: CharSequence?, start: Int, count: Int, after: Int) {
    }


    override fun afterTextChanged(s: Editable?) {
    }
}